src/scripts/mod.rs

```diff
@@ -0,0 +1,318 @@
+use std::fs::{self, File};
+use std::io::{self, prelude::*};
+use std::net::IpAddr;
+use std::path::PathBuf;
+use subprocess::{Exec, ExitStatus};
+use text_placeholder::Template;
+
+static DEFAULT: &'static str = r#"tags = ["core_approved", "RustScan", "default"]
+developer = [ "RustScan", "https://github.com/RustScan" ]
+join_ports = ","
+call_format = "nmap -vvv -p {{port}} {{ip}}"
+"#;
+
+pub fn init_scripts(scripts: String) -> Result<Vec<ScriptFile>> {
+    let mut scripts_to_run: Vec<ScriptFile> = Vec::new();
+
+    match scripts.as_ref() {
+        "none" => Ok(scripts_to_run),
+        "default" => {
+            let default_script = toml::from_str::<ScriptFile>(&DEFAULT).unwrap();
+            scripts_to_run.push(default_script);
+            Ok(scripts_to_run)
+        }
+        "custom" => {
+            let scripts_dir_base = match dirs::home_dir() {
+                Some(dir) => dir,
+                None => return Err(anyhow!("Could not infer scripts path.")),
+            };
+            let script_paths = match find_scripts(scripts_dir_base) {
+                Ok(script_paths) => script_paths,
+                Err(e) => return Err(anyhow!(e)),
+            };
+            debug!("Scripts paths \n{:?}", script_paths);
+
+            let parsed_scripts = parse_scripts(script_paths);
+            debug!("Scripts parsed \n{:?}", parsed_scripts);
+
+            let script_config = match ScriptConfig::read_config() {
+                Ok(script_config) => script_config,
+                Err(e) => return Err(anyhow!(e)),
+            };
+            debug!("Script config \n{:?}", script_config);
+
+            // Only Scripts that contain all the tags found in ScriptConfig will be selected.
+            if script_config.tags.is_some() {
+                let config_hashset: HashSet<String> =
+                    script_config.tags.unwrap().into_iter().collect();
+                for script in &parsed_scripts {
+                    if script.tags.is_some() {
+                        let script_hashset: HashSet<String> =
+                            script.tags.clone().unwrap().into_iter().collect();
+                        if config_hashset.is_subset(&script_hashset) {
+                            scripts_to_run.push(script.to_owned());
+                        }
+                        debug!(
+                            "\nScript tags does not match config tags {:?} {}",
+                            &script_hashset,
+                            script.path.clone().unwrap().display()
+                        );
+                    }
+                }
+            }
+            debug!("\nScript(s) to run {:?}", scripts_to_run);
+            Ok(scripts_to_run)
+        }
+        &_ => Ok(scripts_to_run),
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Script {
+    // Path to the script itself.
+    path: Option<PathBuf>,
+
+    // Ip got from scanner.
+    ip: IpAddr,
+
+    // Ports found with portscan.
+    ports: Vec<u16>,
```

Comment:
What do you think of calling this `open_ports` since we now know they are open and calling the `port` option something like: `conditional_port` or `port_trigger`? Something that indicates that this port is the condition for the script to run.

Reply:
I changed the ports to open_ports so it's more descriptive, but I'm not sure in the conditional_port or port_trigger. What would you say for predefined_port? 🤔

Reply:
I changed the variable name in the script files and the struct to `trigger_port`, I think that describes it good.